packages/lib-classifier/src/store/utils/ClassificationQueue.js

```diff
@@ -58,33 +60,36 @@ class ClassificationQueue {
     }
 
     if (process.env.NODE_ENV !== 'test') console.log('Saving queued classifications:', pendingClassifications.length)
-    return Promise.all(pendingClassifications.map((classificationData) => {
-      return this.apiClient.post(this.endpoint, { classifications: classificationData })
-        .then((response) => {
-          if (response.ok) {
-            const savedClassification = response.body.classifications[0]
-            if (process.env.NODE_ENV !== 'test') console.log('Saved classification', savedClassification.id)
-            this.onClassificationSaved(savedClassification)
-            this.addRecent(savedClassification)
-          }
-        })
-        .catch((error) => {
-          if (process.env.NODE_ENV !== 'test') console.error('Failed to save a queued classification:', error)
-
-          if (error.status !== 422) {
-            try {
-              this.store(classificationData)
-              if (!this.flushTimeout) {
-                this.flushTimeout = setTimeout(this.flushToBackend.bind(this), RETRY_INTERVAL)
+    return getBearerToken(this.authClient)
+      .then((authorization) => {
+        return Promise.all(pendingClassifications.map((classificationData) => {
+          return this.apiClient.post(this.endpoint, { classifications: classificationData }, authorization)
```

Comment:
The nested Promises here are becoming hard to follow.

Reply:
This could be refactored to use async/await

Reply:
cc #525